packages/accounts-base/accounts_server.js

```diff
@@ -1078,11 +1078,13 @@ Ap._generateStampedLoginToken = function () {
 function expirePasswordToken(accounts, oldestValidDate, tokenFilter, userId) {
   var userFilter = userId ? {_id: userId} : {};
 
-  accounts.users.update(_.extend(userFilter, tokenFilter, {
-    $or: [
-      { "services.password.reset.when": { $lt: oldestValidDate } },
-      { "services.password.reset.when": { $lt: +oldestValidDate } }
-    ]
+  accounts.users.update(_.extend(userFilter, 
+    {$and: [tokenFilter, {
+      $or: [
+        { "services.password.reset.when": { $lt: oldestValidDate } },
+        { "services.password.reset.when": { $lt: +oldestValidDate } }
+      ]
+    }]
```

Comment:
I realize the formatting here wasn't great to begin with, but this could still use some cleaning up - we should take this opportunity to fix it.

At the very least, with this complexity, the `$or` should become a variable:

```js
  const resetRangeOr = {
    $or: [
      "services.password.reset.when": { $lt: oldestValidDate } },
      "services.password.reset.when": { $lt: +oldestValidDate } },
    ],
  };
```

Then, we should fix the indentation and stop `extend`-ing into a variable name which doesn't make sense after it's changed (Since `_.extend` mutates the first parameter, and then it's no longer the `userFilter` once the `tokenFilter` and `resetRange` are in there):

```js
accounts.users.update(
  _.extend(
    {},
    userFilter,
    { $and: [ tokenFilter, resetRangeOr ] }
  )
);
```

Or instead of the last suggestion (though I realize this doesn't match the existing code, it's much more clear):

```js
  // resetRangeOr and userFilter are already defined above.
  const expireFilter = { $and: [tokenFilter, resetRangeOf] };

  accounts.users.update({ ...userFilter, ...expireFilter });
```